include/deal.II/lac/vector.templates.h

```diff
@@ -134,12 +135,62 @@ namespace internal
                            tbb::auto_partitioner());
       }
     else if (vec_size > 0)
-      functor(0,vec_size);
 #endif
+      functor(0,vec_size);
```

Comment:
Well, the patch is already merged, but I'm generally not a fan of having `#ifdef`s interrupt control flow as is done here. It's hard to read code where control flow is different in the `#if` and `#else` branch, and would prefer to copy code as necessary